h/services/subscription.py

```diff
@@ -0,0 +1,85 @@
+from typing import List
+
+from sqlalchemy import func
+from webob.cookies import SignedSerializer
+
+from h.models import Subscriptions
+from h.security import derive_key
+
+
+class InvalidUnsubscribeToken(Exception):
+    ...
+
+
+class SubscriptionService:
+    class Type:
+        REPLY = "reply"
+
+        ALL_TYPES = (REPLY,)
+
+    def __init__(self, db_session, secret, salt):
+        self._db_session = db_session
+        self._token_serializer = SignedSerializer(
+            secret=derive_key(key_material=secret, salt=salt, info=b"h.notification"),
+            salt=None,
+        )
+
+    def get_subscriptions(self, user_id: str) -> dict:
+        return {
+            subscription.type: subscription.active
+            for subscription in self._get_subscriptions(user_id)
+        }
```

Comment:
The format here is a reply like: `{"reply": True}`, but about the only places it's used we seem to want sets where presence in the set indicates subscription.

For me a dict with `True` and `False` seems better because you can iterate the keys even if you aren't subscribed. On the other hand, a set is quite neat in it's own way?